suse_migration_services/units/grub_setup.py

```diff
@@ -32,28 +33,44 @@ def main():
     DistMigration update grub to migrated version
 
     Setup and update grub with content from the migrated system
+    Uninstall activation and live migration packages such that
+    a subsequent call of grub2-mkconfig will delete the migration
+    menu entry
     """
     root_path = Defaults.get_system_root_path()
     grub_config_file = Defaults.get_grub_config_file()
 
     try:
         log.info('Running grub setup service')
-        # uninstall suse-migration-activation so new grub
-        # menu does not have the migration entry
-        log.info('Uninstalling suse-migration-activation')
-        Command.run(
-            [
-                'chroot', root_path,
-                'zypper', '--non-interactive', '--no-gpg-checks',
-                'remove', '-u', 'suse-migration-*-activation'
-            ], raise_on_error=False
-        )
-        log.info('Creating new grub menu with target')
-        Command.run(
-            [
-                'chroot', root_path, 'grub2-mkconfig', '-o',
-                '{0}{1}'.format(os.sep, grub_config_file)
-            ]
+        system_package_list = Command.run(
+            ['rpm', '--root', root_path, '-qa'], raise_on_error=False
+        ).output.split(os.linesep)
+        migration_packages = [
+            package for package in system_package_list if re.match(
+                r'suse-migration-.*-activation|SLE.*-Migration', package
+            )
+        ]
```

Comment:
I think you can simplify the search for the migration packages. Rather then getting a list of all installed packages. We know that the migration packages leave certain files behind, thus we should be able to use

rpm --root .... -qf $A_FILE_FROM_ACTIVATION
rpm --root .... -qf $A_FILE_FROM_MIGRATION

and think of it that way we can simply check this without using rpm, I think. Getting the full list of all package of the system to be migrated could result in a list of thousands of packages. I thinks it's a heavy handed approach.

Reply:
Agree with Robert, if we can avoid check in the configured repos, better as we know we dont have to check there.